plugins/inputs/smart/smart.go

```diff
@@ -817,6 +829,14 @@ func (m *Smart) gatherDisk(acc telegraf.Accumulator, device string, wg *sync.Wai
 					deviceFields[field] = val
 				}
 			}
+
+			// If the attribute name matches on in deviceFieldNames
+			// save the value to a field
+			if field, ok := deviceFieldNames[attr[2]]; ok {
+				if val, err := parseRawValue(attr[4]); err == nil {
+					deviceFields[field] = val
+				}
+			}
```

Comment:
Need to check to make sure the length of `attr` is bigger then 4, the check `len(attr) > 1` happens only in the above for loop but that might not be the case here.

Reply:
Yep, nice catch - thanks